api/datasets.go

```diff
@@ -497,6 +410,141 @@ func (h *DatasetHandlers) initHandler(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(env)
 }
 
+func formFileDataset(dsp *dataset.DatasetPod, r *http.Request) (cleanup func(), err error) {
+	var rmFiles []*os.File
+	cleanup = func() {
+		for _, f := range rmFiles {
+			// TODO - log error?
+			os.Remove(f.Name())
```

Comment:
This line is kind of terrifying. Manual inspection convinces me that it's safe, but I can imagine a scenario in the future where that's not the case. It appears that, currently, all elements added to rmFiles are in the temporary directory, which means their paths should start with the value returned by "TempDir". How about, outside this loop, get the value of TempDir, then only remove files that match this prefix.

Reply:
Oh man you're so right. This is like, way under-considered. We're on a mad time crunch so I'm going to slap a huge TODO on this & we should fix this in a patch by Friday. Thankfully this won't to anything bad in it's current incarnation, but totally agreed on the dangers of this kind of code down the road #rememberUltima